### ISSUE TITLE
Regression with parent-child, object type and aggregations?

### ISSUE THREAD

**Issue by paolociccarese** [elastic/elasticsearch#12164]
I am experiencing a regression when using parent-child relationships and aggregations of 
inner objects in Elasticsearch versions newer than 1.4.4.

Here is a simple made-up mapping that can demonstrate my issue:

```
curl -XDELETE 'http://localhost:9200/test'
curl -XPOST 'localhost:9200/test' -d '
{
  "mappings": {
  "Gym": {
      "properties": {
        "GymUID": {
          "type": "string",
          "include_in_all": true,
          "index": "not_analyzed"
        },
        "Short Description": {
          "type": "string"      
        }
       }
    },
    "Member": {
      "_parent": {
        "type": "Gym"
      },
      "properties": {
        "MemberUID": {
          "type": "string",
          "include_in_all": true,
          "index": "not_analyzed"
        },
        "Age": {
          "type": "integer",
          "null_value": 0
        },
        "Gender": {
          "type": "string",
          "include_in_all": true,
          "index": "not_analyzed"
        }
     }
     },
     "Visit": {
      "_parent": {
        "type": "Member"
      },
      "properties": {
        "Location": {
          "properties": {
            "Name": {
              "type": "string",
              "include_in_all": true,
              "index": "not_analyzed"
            }
          }
        },
        "Member": {
          "properties": {
            "MemberUID": {
              "type": "string",
              "include_in_all": true,
              "index": "not_analyzed"
            },
            "Age": {
              "type": "integer",
              "null_value": 0
            },
            "Gender": {
              "type": "string",
              "include_in_all": true,
              "index": "not_analyzed"
            }
          }
        },
        "VisitUID": {
          "type": "string",
          "include_in_all": true,
          "index": "not_analyzed"
        }
      }
    }
  }
}
'
```

We can now post some data
```
curl -XPOST 'http://localhost:9200/test/Gym/Gym1' -d '{"GymUID":1,"Short Description":"Sport center"}'
curl -XPOST 'http://localhost:9200/test/Member/1?parent=Gym1&routing=Gym1' -d '{"MemberUID":"1","Gender":"Male","Age":65}'
curl -XPOST 'http://localhost:9200/test/Visit/1?parent=1&routing=Gym1' -d '{"VisitUID":"1","Location":{"Name":"Paolo Ciccarese house"},"Member":{"MemberUID":"1","Gender":"Male","Age":65}}'
```

If now we query for visits of Male members and aggregate by age:
```
curl -XGET 'http://localhost:9200/test/Visit/_search?pretty=true&size=1' -d '
{
  "query":{
     "match":{"Member.Gender":"Male"} 
  },
  "aggs":{
     "members":{"terms":{"field":"Member.Age"}}
  }
}'
```

The query returns one result and an empty aggregation for versions after 1.4.4 (I've tried 1.5.2 and 1.6.0).
It returns one result and one item in the aggregation bucket with version 1.4.4.

Notice that if I use the same mapping for Visit but without the parents:
```
curl -XDELETE 'http://localhost:9200/test'
curl -XPOST 'localhost:9200/test' -d '
{
  "mappings": {
     "Visit": {
      "properties": {
        "Location": {
          "properties": {
            "Name": {
              "type": "string",
              "include_in_all": true,
              "index": "not_analyzed"
            }
          }
        },
        "Member": {
          "properties": {
            "MemberUID": {
              "type": "string",
              "include_in_all": true,
              "index": "not_analyzed"
            },
            "Age": {
              "type": "integer",
              "null_value": 0
            },
            "Gender": {
              "type": "string",
              "include_in_all": true,
              "index": "not_analyzed"
            }
          }
        },
        "VisitUID": {
          "type": "string",
          "include_in_all": true,
          "index": "not_analyzed"
        }
      }
    }
  }
}
'
```

And I push the same data
```
curl -XPOST 'http://localhost:9200/test/Visit/1' -d '{"VisitUID":"1","Location":{"Name":"Paolo Ciccarese house"},"Member":{"MemberUID":"1","Gender":"Male","Age":65}}'
```

The same query works beautifully:
```
curl -XGET 'http://localhost:9200/test/Visit/_search?pretty=true&size=1' -d '
{
  "query":{
     "match":{"Member.Gender":"Male"} 
  },
  "aggs":{
     "members":{"terms":{"field":"Member.Age"}}
  }
}'
```

and returns one item in the aggregation bucket no matter which version I use.

Any idea?

**Comment by clintongormley** [comment]
Hi @paolociccarese 

You're being bitten by ambiguous field resolution.  It is using the `Age` field in the `Member` type, instead of the `Member.Age` field in the `Visit` type.

We have solved these issues in 2.0 with #8871 

However in 1.6 you can do the following:

    GET /test/Visit/_search?pretty=true&size=1
    {
      "query":{
         "match":{"Member.Gender":"Male"} 
      },
      "aggs":{
         "members":{"terms":{"field":"Visit.Member.Age"}}
      }
    }

**Comment by pciccarese** [comment]
Thank you @clintongormley !
